internal/envconfig/envconfig.go

```diff
@@ -25,25 +25,26 @@ import (
 	"strings"
 )
 
-const (
-	prefix                  = "GRPC_GO_"
-	txtErrIgnoreStr         = prefix + "IGNORE_TXT_ERRORS"
-	advertiseCompressorsStr = prefix + "ADVERTISE_COMPRESSORS"
-)
-
 var (
 	// TXTErrIgnore is set if TXT errors should be ignored ("GRPC_GO_IGNORE_TXT_ERRORS" is not "false").
-	TXTErrIgnore = !strings.EqualFold(os.Getenv(txtErrIgnoreStr), "false")
+	TXTErrIgnore = boolFromEnv("GRPC_GO_IGNORE_TXT_ERRORS", true)
 	// AdvertiseCompressors is set if registered compressor should be advertised
 	// ("GRPC_GO_ADVERTISE_COMPRESSORS" is not "false").
-	AdvertiseCompressors = !strings.EqualFold(os.Getenv(advertiseCompressorsStr), "false")
+	AdvertiseCompressors = boolFromEnv("GRPC_GO_ADVERTISE_COMPRESSORS", true)
 	// RingHashCap indicates the maximum ring size which defaults to 4096
 	// entries but may be overridden by setting the environment variable
 	// "GRPC_RING_HASH_CAP".  This does not override the default bounds
 	// checking which NACKs configs specifying ring sizes > 8*1024*1024 (~8M).
 	RingHashCap = uint64FromEnv("GRPC_RING_HASH_CAP", 4096, 1, 8*1024*1024)
 )
 
+func boolFromEnv(envVar string, def bool) bool {
+	if def {
+		return !strings.EqualFold(os.Getenv(envVar), "false")
+	}
+	return strings.EqualFold(os.Getenv(envVar), "true")
+}
```

Comment:
Would it be better to use `os.LookupEnv()` here? I feel this is a little easier to read. What do you think?

```golang
func boolFromEnv(envVar string, def bool) bool {
	val, ok := os.LookupEnv(envVar)
	if !ok {
		return def
	}
	return strings.EqualFold(val, "true")
}
```

Reply:
This isn't right, though?  The default value affects whether we require "true" or "false" to go against the default.  I.e. if the default is `true` then we will return `true` from this function unless the env var contains `"false"` (case-insensitive).

Reply:
Could we add a docstring clarifying this. Thanks.

Reply:
Done!